002-tidyr-stringr.Rmd

```diff
@@ -0,0 +1,133 @@
+---
+output: github_document
+---
+
+# Tidyup 2: tidyr \<-> stringr alignment
+
+**Champion**: Hadley Wickham  
+**Co-champion**: Mine Çetinkaya-Rundel  
+**Status**: Draft
+
+## Abstract
+
+tidyr provides a handful functions like `separate()` and `extract()` that perform string manipulation.
+Unfortunately, these functions are only weakly connected to their stringr equivalents, making them harder to learn and remember.
+This tidyup proposes a new cohesive family of string manipulation functions to tackle common problems that arise during data tidying.
+The existing functions will remain so existing code is unaffected, but will be superseded to steer folks towards a more consistent API.
+
+```{r setup}
+library(tidyr)
+library(stringr)
+```
+
+## Motivation
+
+stringr and tidyr provide surprisingly different interfaces for string manipulation.
+This means that common tasks require integrating two distinct syntaxes and learning more about stringr doesn't improve your use of tidyr.
+
+The root cause of the problem is that stringr was designed before I understood the importance of data frames, so many stringr functions return matrices (or lists of matrices), which makes them hard to use with other tidyverse packages.
+No stringr function returns a data frame, which is particularly painful given that `summarise()` and `mutate()` can now use data frame results to add multiple columns.
```

Comment:
Is the issue that stringr functions don't return data frames? Or that they don't return vectors? I might be thinking about this wrong but forcats functions also don't return data frames, for example, but they're a lot easier to use in pipelines than string functions.

Reply:
I think it's something like some stringr functions need to return multiple "things", and in that case it's now obvious that you should be returning a data frame instead of a matrix.